### ISSUE TITLE
Nullable analysis treats non-nullable `field`-backed property as possibly null at use site

### ISSUE THREAD

**Issue by DoctorVanGogh** [dotnet/roslyn#84957]
# Version

Observed with:

C# ~~13~~14 field-backed properties (field)
Nullable reference types enabled
```
.NET SDK:
 Version:           10.0.302
 Commit:            35b593bebf
 Workload version:  10.0.300-manifests.714b12c0
 MSBuild version:   18.6.11+35b593beb
```

# Description
A non-nullable property implemented using a field-backed property getter is reported as possibly null at the *use site*, even though:

 - The property's declared type is non-nullable.
 - The getter implementation returns a non-null value.
 - The getter implementation *itself* does not show any null assignment/return warning
 - An equivalent implementation using an explicit backing field does not produce the warning.

## Repro
```cs
    internal class Foo {
        
        private Bar? _bang1;

        public Foo() {  // CS8618: Wiz2/Wiz3/Wiz4/Wiz5 not definitely assigned
            Wiz1 = Bang1;  // no warning
            Wiz2 = Bang2;  // CS8601
            Wiz3 = Bang3;  // CS8601
            Wiz4 = Bang4;  // CS8601
            Wiz5 = Bang5;  // CS8601
        }




        public Bar Wiz1 { get; set; }
        public Bar Wiz2 { get; set; }
        public Bar Wiz3 { get; set; }
        public Bar Wiz4 { get; set; }
        public Bar Wiz5 { get; set; }

        public Bar Bang1 => _bang1 ??= new Bar();

        public Bar Bang2 => field ??= new Bar();

        public Bar Bang3 => (field ??= new Bar())!;

        public Bar Bang4 {
            get {
                field ??= new Bar();
                return field;
            }
        }

        public Bar Bang5 {
            get {
                field = new Bar();
                return field;
            }
        }
    }

    public class Bar { };
```
### Actual Behavior
Assignment from `field` backed non-null properties produces:
```
CS8601: Possible null reference assignment.
```
The warning is reported at
```
WizX = BangX;
```

### Expected Behavior
No warning should be produced.
All `field` backed `BangX` properties are declared as `public Bar BangX` which is non nullable. Consumers should be able to rely on that contract.
Additionally, the implementation(s) are all non null, do not themselves warn about possible null value returns and the equivalent explicit-backing-field implementation does not produce any warning.

Something is not right here.

### Diagnostic Id
`CS8601: Possible null reference assignment`.

---
The nullable analysis behavior is inconsistent between properties with legacy, manually defined backing fields and `field` keyword backing fields.
The analysis is inconsistent between the property getter itself (correctly showing no warnings) and *uses* of the property (where purported warnings suddenly show).
Neither the property's declared nullability contract nor the getter implementation indicate that a nullable value may be returned. Nevertheless, use sites are analyzed as though the property were nullable.

**Comment by RikkiGibson** [comment]
Thanks for reporting. This looks like a dupe of #77991. I think we can prioritize this.
